ui_tests/conftest.py

```diff
@@ -0,0 +1,17 @@
+import pytest
+from django.core.management import call_command
+from pytest_django.plugin import DjangoDbBlocker
+
+# pylint: disable=unused-argument
+
+
+@pytest.fixture(scope="session")
+def load_test_data(django_db_setup: None, django_db_blocker: DjangoDbBlocker) -> None:
+    """
+    Load the test data initially for all test cases
+
+    :param django_db_setup: The fixture providing the database availability
+    :param django_db_blocker: The fixture providing the database blocker
+    """
+    with django_db_blocker.unblock():
+        call_command("loaddata", "integreat_cms/cms/fixtures/test_data.json")
```

Comment:
*This is suggestion is basically just me noodling around when playing with this PR, trying to find what kind of fixtures could make sense. I stopped at writing some to auto-login to any role and starting the test on the dashboard (which also serves as a check to make sure login was successful, because if that assertion fails the test shows as `ERROR` instead of `FAILED` and might save time debugging), plus some comfort fixtures like `page_as_root`, `page_as_*` etc. for all roles so it's easy to test individual roles without having to know that the name for the relevant fixture to parametrize is `login_role_user_cookies` (since that name is not meant to be used directly by tests):*

```python
import pytest
from playwright.sync_api import expect, Page
from ui_tests.conftest import ROOT, AUTHOR

def test_something(page_as_root: Page, page_as_author: Page) -> None:
    page = page_as_root if page_as_root is not None else page_as_author
    # …

# …is more or less equivalent to…

@pytest.mark.parametrize("login_role_user_cookies", [ROOT, AUTHOR], indirect=True)
def test_something(page_as_role: Page) -> None:
    page, role = page_as_role
    # …
```

*This also contains a few small general improvements, like adding a module docstring in the same style as `tests/conftest.py`, importing the roles, role groups and the `load_test_data` fixture from there.*

*There are two TODOs I left in here, but I tried to take care to otherwise clean up and document my work so it is neither lost nor causes way too much unnecessary additional work later, if you someone else were to try and include it in a PR and had to understand it in depth in order to satisfy pylint and co.*

```suggestion
"""
This module contains shared fixtures for pytest, specific to ui tests
"""

from __future__ import annotations

from typing import TYPE_CHECKING

import pytest
from django.contrib.auth import get_user_model
from django.test.client import Client
from playwright.sync_api import BrowserContext, BrowserType, expect, Page
from pytest_django.live_server_helper import LiveServer

# pylint: disable=unused-import
from tests.conftest import ALL_ROLES, ANONYMOUS, APP_TEAM, AUTHOR, CMS_TEAM, EDITOR, EVENT_MANAGER, HIGH_PRIV_STAFF_ROLES, load_test_data, MANAGEMENT, MARKETING_TEAM, OBSERVER, PRIV_STAFF_ROLES, REGION_ROLES, ROLES, ROOT, SERVICE_TEAM, STAFF_ROLES, WRITE_ROLES

if TYPE_CHECKING:
    from typing import Generator, Sequence

    from _pytest.fixtures import SubRequest
    from playwright._impl._api_structures import SetCookieParam
    from pytest_django.plugin import DjangoDbBlocker  # type: ignore[attr-defined]


# pylint: disable=redefined-outer-name
@pytest.fixture(scope="session")
def login_role_user_cookies(
    request: SubRequest, django_db_setup: None, load_test_data: None, django_db_blocker: DjangoDbBlocker, live_server: LiveServer # noqa: F811
) -> tuple[list[dict[str,str]], str]:
    """
    Get the test user of the current role, force a login and extract the session cookies. Gets executed only once per user.

    :param request: The request object providing the parametrized role variable through ``request.param``
    :param django_db_setup: The fixture providing the database availability
    :param load_test_data: The fixture providing the test data (see :meth:`~tests.conftest.load_test_data`)
    :param django_db_blocker: The fixture providing the database blocker
    :param live_server: The fixture providing a running full blown server instance to test against
    :return: A tuple consisting of the cookies representing the session for the respective user and the name of the associated role
    """
    client = Client()
    cookies = []
    # Only log in user if the role is not anonymous
    if request.param != ANONYMOUS:
        with django_db_blocker.unblock():
            user = get_user_model().objects.get(username=request.param.lower())
            client.force_login(user)

        domain = f"{live_server.thread.host}:{live_server.thread.port}"
        for name, properties in client.cookies.items():
            cookie = {
                "name": name,
                "value": properties.value, # needs to be put in explicitly!
                "domain": domain,
                "path": properties["path"],
            }
            cookies.append(cookie)

    return cookies, request.param


@pytest.fixture(scope="function")
def login_role_user(login_role_user_cookies: tuple[Sequence[SetCookieParam], str], browser_type: BrowserType, django_db_blocker: DjangoDbBlocker) -> Generator[tuple[BrowserContext, str], None, None]:
    """
    Launch a fresh browser context with a valid session for the current user.

    :param login_role_user_cookies: A tuple of the session cookies and the name of the associated user role
    :param browser_type: The object to launch a browser context
    :param django_db_blocker: The fixture providing the database blocker
    :return: The browser context and the current role
    """
    cookies, role = login_role_user_cookies

    # TODO: expose headless through parametrization
    #       and set it to False
    #       - whenever this parametrization tells us to
    #       - OR tests were invoked with a global non-headless flag
    browser = browser_type.launch(headless=False)
    # TODO: enable changing TZ and locale through parametrization
    #context = browser.new_context(timezone_id="Europe/Berlin", locale="de-DE")
    context = browser.new_context(timezone_id="Europe/Berlin", locale="en-GB")
    context.add_cookies(cookies)

    yield context, role

    context.close(reason="Test finished")


@pytest.fixture(scope="function")
def page_as_role(login_role_user: tuple[BrowserContext, str], live_server: LiveServer) -> Page:
    """
    Launch a fresh browser context with a valid session for the current user.

    :param login_role_user: The browser context and the current role
    :param live_server: The fixture providing a running full blown server instance to test against
    :return: The page on the default page (login page for anonymous, otherwise the respective dashboard)
    """
    context, role = login_role_user
    page = context.new_page()
    page.goto(live_server.url)
    if role == ANONYMOUS:
        expect(page.get_by_role("heading", name="Login")).to_be_visible()
    else:
        expect(page.get_by_role("heading", name="Login")).to_be_hidden()
        name = "Admin Dashboard" if role in STAFF_ROLES else "My Dashboard"
        expect(page.get_by_role("heading", name=name)).to_be_visible()

    return page


ROLE_FIXTURES = [f"page_as_{role.lower()}" for role in ALL_ROLES]

def pytest_generate_tests(metafunc: pytest.Metafunc) -> None:
    """
    Hook called by pytest on collection to generate tests (e.g. dynamically adding parametrizations for fixtures).
    We use this instead of defining default parametrizations on `login_role_user_cookies` directly
    in order to implement the page_as_<ROLE> fixtures as syntactic sugar.

    :param metafunc: The object describing the test and details like attached fixtures
    """
    if "login_role_user_cookies" in metafunc.fixturenames:
        already_parametrized = None
        for mark in metafunc.definition.iter_markers(name="parametrize"):
            if mark.args[0] == "login_role_user_cookies":
                already_parametrized = mark
                # Theoretically, we could choose to overwrite the given parametrization later like this:
                # mark.args[1] = (mark.args[1][0], REPLACEMENT_PARAMETERS)
                # because mark.args[1] is a tuple of the name of the fixture the parameters are for, and the parameters themselves.
                # But in this case we probably don't want to overwrite it if it is already parametrized.

        if not already_parametrized:
            # The fixtures like page_as_root don't do anything by themselves
            # except for depending on page_as_role and passing its result without modification.
            # The actual logic happens here, where we parametrize the login_role_user_cookies fixture,
            # which is the one actually doing something with the parameters in the end,
            # by simply checking which page_as_* fixtures are attached to the test.
            # If none are found, we just parametrize it with ALL_ROLES. This is necessary because
            # we cannot decorate the login_role_user_cookies fixture as we would do usually with
            # @pytest.fixture(scope="session", params=ALL_ROLES)
            # because it seems then we cannot distinguish whether that parametrization stems from the fixture definition
            # or whether it was parametrized on the test. (see how we determined already_parametrized above)
            found_roles = [role for role in ALL_ROLES if f"page_as_{role.lower()}" in metafunc.fixturenames]
            if not found_roles:
                found_roles = ALL_ROLES
            metafunc.parametrize("login_role_user_cookies", found_roles, indirect=True)


# Create the dummy page_as_* fixtures
for name in ROLE_FIXTURES:
    def fixture_func(page_as_role: Page) -> Page:
        """
        The page on the default page (login page for anonymous, otherwise the respective dashboard).

        :param page_as_role: The page on the default page, logged in
        :return: The page on the default page, logged in
        """
        return page_as_role

    globals()[name] = pytest.fixture(scope="function")(fixture_func)
```